CallPlayAudio/CallPlayTerminate.cs

```diff
@@ -0,0 +1,173 @@
+            reportCancellationToken = reportCancellationTokenSource.Token;
+
+            try
+            {
+                callConnection = await CreateCallAsync(targetPhoneNumber).ConfigureAwait(false);
+
+                await PlayAudioAsync().ConfigureAwait(false);
+              
+                // Wait for the call to terminate
+                await callTerminatedTask.Task.ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogMessage(Logger.MessageType.ERROR, $"Call ended unexpectedly, reason: {ex.Message}");
+            }
+        }
+
+        private async Task<CallConnection> CreateCallAsync(string targetPhoneNumber)
+        {
+            try
+            {
+                //Preparting request data
+                CallSource source = new CallSource(new CommunicationUserIdentifier(callConfiguration.SourceIdentity));
+                source.CallerId = new PhoneNumberIdentifier(callConfiguration.SourcePhoneNumber);
+                var target = new PhoneNumberIdentifier(targetPhoneNumber);
+
+                var createCallOption = new CreateCallOptions(source,
+                    new List<CommunicationIdentifier>() { target },
+                    new Uri(callConfiguration.AppCallbackUrl));
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Performing CreateCall operation");
+                var call = await callClient.CreateCallAsync(createCallOption).ConfigureAwait(false);
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"CreateCallConnectionAsync response --> {call.GetRawResponse()}, Call Connection Id: { call.Value.CallConnection.CallConnectionId}");
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"Call initiated with Call Connection id: { call.Value.CallConnection.CallConnectionId}");
+
+                RegisterToCallStateChangeEvent(call.Value.CallConnection.CallConnectionId);
+
+                //Wait for operation to complete
+                await callEstablishedTask.Task.ConfigureAwait(false);
+
+                return call.Value.CallConnection;
+            }
+            catch (Exception ex)
+            {
+                Logger.LogMessage(Logger.MessageType.ERROR, string.Format("Failure occured while creating/establishing the call. Exception: {0}", ex.Message));
+                throw ex;
+            }
+        }
+
+        private async Task PlayAudioAsync()
+        {
+            if (reportCancellationToken.IsCancellationRequested)
+            {
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Cancellation request, PlayAudio will not be performed");
+                return;
+            }
+
+            try
+            {
+                // Preparing data for request
+                var playAudioRequest = new PlayOptions()
+                {
+                    OperationContext = Guid.NewGuid().ToString(),
+                    Loop = true,
```

Comment:
Lets keep the flag to false and once the playing of the audio is completed  we should await for PlayCompleted event to arrive on the callback event. 
I dont see that we are waiting for PlayCompleted event here ?

Reply:
Once we confirm that we got back the PlayCompleted event we should hangup the call